pkg/container/docker_runner.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2022 Chainguard, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package container
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/stdcopy"
+)
+
+type DKRunner struct {
+	Runner
+}
+
+// DockerRunner returns a Docker Runner implementation.
+func DockerRunner() Runner {
+	return &DKRunner{}
+}
+
+// StartPod starts a pod for supporting a Docker task, if
+// necessary.
+func (dk *DKRunner) StartPod(cfg *Config) error {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return err
+	}
+	defer cli.Close()
+
+	mounts := []mount.Mount{}
+	for _, bind := range cfg.Mounts {
+		mounts = append(mounts, mount.Mount{
+			Type:   mount.TypeBind,
+			Source: bind.Source,
+			Target: bind.Destination,
+		})
+	}
```

Comment:
Yes, basically.  The only question I haven't answered is how we pull the data back down from the cluster in order to package it into an apk.

Reply:
We could `kubectl cp` or the equivalent `kubectl exec tar` gloop to pipe the file contents back over stdout. Both require `tar` to be present in the container, but that's okay. That also fairly nicely solves the problem of giving the build container credentials to push to GCS directly.

Reply:
Well, any container managed by Melange can't push to GCS anyway, because the host needs to process the build result in order to generate and sign the apk, but yeah :)